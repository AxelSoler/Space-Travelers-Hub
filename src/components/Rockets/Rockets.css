.rocket-holder {
  display: flex;
  gap: 1.5rem;
  margin-bottom: 1rem;
  padding: 0 2rem;
}

.img-holder img {
  width: 250px;
}

.data-rocket {
  flex: 1;
}
