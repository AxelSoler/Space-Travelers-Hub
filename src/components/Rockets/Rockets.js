import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux/es/exports';
import './Rockets.css';
import { bookRocket } from '../../redux/rockets/rockets';

const Rockets = (props) => {
  const {
    name, img, description, id,
  } = props;

  const [isActive, setIsActive] = useState(false);

  const dispatch = useDispatch();

  const bookRocketHandler = () => {
    dispatch(bookRocket(id));
  };

  const displayHandler = () => {
    setIsActive((current) => !current);
  };

  return (
    <div className="rocket-holder">
      <div className="img-holder">
        <img src={img} alt="rocket" />
      </div>
      <div className="data-rocket">
        <div className="name-holder">{name}</div>
        <div className="description-holder">{description}</div>
        <button
          type="button"
          className={isActive ? 'no-active' : 'active'}
          onClick={() => {
            bookRocketHandler();
            displayHandler();
          }}
        >
          Reserve Rocket
        </button>
        <button
          type="button"
          className={isActive ? 'active' : 'no-active'}
          onClick={displayHandler}
        >
          Cancel Reservation
        </button>
      </div>
    </div>
  );
};

Rockets.propTypes = {
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};

export default Rockets;
