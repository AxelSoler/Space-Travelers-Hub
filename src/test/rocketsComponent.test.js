import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rockets from '../components/Rockets/Rockets';
describe('Rockets', () => {
  it('Renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Rockets />
          </Provider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})