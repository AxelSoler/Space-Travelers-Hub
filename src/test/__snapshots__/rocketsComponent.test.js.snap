// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets Renders correctly 1`] = `
<div
  className="rocket-holder"
>
  <div
    className="img-holder"
  >
    <img
      alt="rocket"
    />
  </div>
  <div
    className="data-rocket"
  >
    <div
      className="name-holder"
    />
    <div
      className="description-holder"
    >
      <div>
        <p />
      </div>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
